### PR TITLE
regen/embed.pl: Set flag to avoid rematching pattern

### PR DESCRIPTION
**Pull request by khwilliamson**

### CODE REVIEW

File: regen/embed.pl [hunk 1]
```diff
@@ -110,7 +110,7 @@ sub generate_proto_h {
         my $has_context = ( $flags !~ /T/ );
         my $never_returns = ( $flags =~ /r/ );
         my $binarycompat = ( $flags =~ /b/ );
-        my $commented_out = ( $flags =~ /m/ );
+        my $has_mflag = ( $flags =~ /m/ );
```
Comment (tonycoz) [inline regen/embed.pl:127]: I think `$is_macro` would be a better name, to better match the other "purpose of flag" names.

Comment (khwilliamson) [reply regen/embed.pl:127]: The problem is that other flags imply that it is a macro  But now that I think about it, it is probably best if those flags cause `m` to be added behind the scenes when it isn't actually present

**Comment by jkeenan** [comment]
@khwilliamson, can we get an update on the status of this p.r.?  Thanks.

**Comment by khwilliamson** [comment]
It's deferred until changes elsewhere enable it

**Comment by khwilliamson** [comment]
I looked at this more, and `is_macro` isn't appropriate because the `M` flag also means it is a macro, and the code really wants `m` as distinct from `M`

**Pull request merged**